### PR TITLE
Remove `RoadwayObjectsWorker` class

### LINKED ISSUE #2144: Update `roadway_objects` package structure
### Summary

The `roadway_objects` package uses the older node/worker package architecture that is unnecessary for ROS 2 packages. The current structure significantly reduces readability without significant gains in other areas. This structure should be simplified.

### Reasoning for new functionality

_No response_

### PR DESCRIPTION
**Pull request by adamlm**
# PR Details
## Description

This PR removes the `RoadwayObjectsWorker` class and migrates its functionality directly into the `RoadwayObjectsNode` class. The previous node/worker architecture implementation was a bit clunky and made the code harder to read. Additionally, the unit tests as implemented were more of component/integration tests. This PR replaces those unit tests with proper component tests using the `launch_testing` framework.

## Related GitHub Issue

Closes #2144 

## Related Jira Key

CDAR-349

## Motivation and Context

The node/worker architecture was a holdover from the ROS 1 implementations, and it does not integrate so well with ROS 2 conventions of subclassing `rclcpp::Node`. In this package, the callback structure bounced between work and node, making it difficult to comprehend the code flow.

## How Has This Been Tested?

The worker's unit tests have been replaced with launch tests (integration/component tests). The replacement tests should have test-parity with the originals.

## Types of changes

- [x] Breaking change (fix or feature that cause existing functionality to change). **Note: This functionality change is internal to the Node. It does not affect other code.**
- [x] Code refactoring

## Checklist:

- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: roadway_objects/test/roadway_objects_empty_map_launch_test.py
Comment (JonSmet) [inline roadway_objects/test/roadway_objects_empty_map_launch_test.py:48]: Just to check my understanding, are we setting up a client for `node_under_test/change_state` here so that TestHarnessNode will not create its publisher until node_under_test is active?

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: roadway_objects/test/roadway_objects_regular_map_launch_test.py
Comment (adamlm) [reply roadway_objects/test/roadway_objects_regular_map_launch_test.py:258]: This is getting initialized. It will create a `LenIncreases` instance that gets passed into the function. In C++ terms, this would be a function object.

File: roadway_objects/test/roadway_objects_regular_map_launch_test.py
Comment (adamlm) [reply roadway_objects/test/roadway_objects_regular_map_launch_test.py:262]: Good point. I will make the fix.

File: roadway_objects/test/roadway_objects_empty_map_launch_test.py
Comment (adamlm) [reply roadway_objects/test/roadway_objects_empty_map_launch_test.py:48]: The `change_state_client` is actually from an older version of the implementation and should be removed. State transitions are now handled by the `try_transition_node`, which spins up a temporary client to do the same thing. This avoids test creators from having to constantly add the transition client to their test harnesses.

File: roadway_objects/test/roadway_objects_regular_map_launch_test.py
Comment (JonSmet) [reply roadway_objects/test/roadway_objects_regular_map_launch_test.py:249]: Makes perfect sense, thanks

**Review by JonSmet — APPROVED** [review]
Looks good, thanks! I'll wait for CI to complete before merging.

**Pull request merged**
